### ISSUE TITLE
Please drop pymacaroons-nacl dependency and use pymacaroons >= 0.13 instead

### ISSUE THREAD

**Issue by andrewshadura** [matrix-org/synapse#3164]
Pymacaroons have switched to nacl recently, so there's no need to depend on pymacaroons-nacl. The upstream pymacaroons should do just fine.

**Comment by andrewshadura** [comment]
Please also see [issue #27 in the synapse packaging](/matrix-org/package-synapse-debian/issues/27).

**Comment by richvdh** [comment]
Until debian stable ships with pymacaroons 0.13, that's not really a thing we can do :/

**Comment by andrewshadura** [comment]
Hrm, why didn’t you comment on the matter at all? You’re not using Debian’s pymacaroons at all, you’re rolling your own package, so you could just as well ship pymacaroons 0.13 in your repository.

**Comment by andrewshadura** [comment]
And also this bug is *not* about the Debian packaging. It’s about Synapse, what ships in Debian stable bears no relevance here.

**Comment by richvdh** [comment]
What ships in debian stable is relevant because if we update synapse to require a particular version of a python package which isn't in debian stable, then we have to package and maintain that python package for debian as well, which is overhead we can do without. There are a whole bunch of things we'd love to upgrade (chiefly Twisted) but have avoided doing so for compatibility with stretch. 

However, I hadn't realised we are already shipping our own pymacaroons deb. In that case, as you note, we could just upgrade it.

**Comment by andrewshadura** [comment]
But you already package and maintain a separate fork of that package, so how is that different from just uploading a package already package for Debian stretch into your repository?

Also, is Debian the only supported OS for Synapse? If not, then it should be relevant for Synapse development as such, since it is a separate issue with packaging.

**Comment by richvdh** [comment]
Debian (and Ubuntu, by proxy) is the only distro we ship pacakages for. Others are expected to install via pip.

**Comment by richvdh** [comment]
The main thing stopping me doing this is that I can't face figuring out how to build a deb of a recent python-pymacaroons. If anybody can provide an idiot's guide, this would have a better chance of being fixed...

**Comment by andrewshadura** [comment]
```
dget -d http://deb.debian.org/debian/pool/main/p/pymacaroons/pymacaroons_0.13.0-1.dsc
cd pymacaroons-0.13.0/
dpkg-buildpackage
```
:slightly_smiling_face:

**Comment by andrewshadura** [comment]
Or if you have *a bit less* generic questions, feel free to ask

**Comment by erikjohnston** [comment]
Fixed in #4399
